semgrep-core/src/engine/Match_rules.ml

```diff
@@ -137,10 +137,10 @@ let error env msg =
   let err = E.mk_error_loc loc (E.MatchingError s) in
   Common.push err env.errors
 
-let (xpatterns_in_formula : S.sformula -> R.xpattern list) =
+let (xpatterns_in_formula : S.sformula -> (R.xpattern * R.inside option) list) =
```

Comment:
Could we make R.inside part of the R.xpattern type? Or have a type with both?

Reply:
That was the first thing I tried, but that would mean a cascade of changes in `Rule.ml` and `Parse_rule.ml`. I would probably then change the `Rule.formula_old` type too.  If we do that I would prefer to do it in a separate clean-up PR.